nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala

```diff
@@ -0,0 +1,116 @@
+package scala.scalanative
+package posix.sys
+
+import native._
+import types.{mode_t, off_t}
+
+// http://pubs.opengroup.org/onlinepubs/9699919799/
+
+@extern
+object mman {
+
+  def mlock(addr: Ptr[Byte], len: CSize): CInt = extern
+
+  def mlockall(flags: CInt): CInt = extern
+
+  def mmap(addr: Ptr[Byte],
+           len: CSize,
+           prot: CInt,
+           flags: CInt,
+           fd: CInt,
+           offset: off_t): Ptr[Byte] = extern
+
+  def mprotect(addr: Ptr[Byte], len: CSize, prot: CInt): CInt = extern
+
+  def msync(addr: Ptr[Byte], len: CSize, flags: CInt): CInt = extern
+
+  def munlock(addr: Ptr[Byte], len: CSize): CInt = extern
+
+  def munlockall(): CInt = extern
+
+  def munmap(addr: Ptr[Byte], len: CSize): CInt = extern
+
+  def posix_madvise(addr: Ptr[Byte], len: CSize, advice: CInt): CInt = extern
+
+  def posix_mem_offset(addr: Ptr[Byte],
+                       len: CSize,
+                       off: Ptr[off_t],
+                       contig_len: Ptr[CSize],
+                       fildes: Ptr[CInt]): CInt = extern
+
+  def posix_typed_mem_get_info(fildes: CInt,
+                               info: Ptr[posix_typed_mem_info]): CInt = extern
+
+  def posix_typed_mem_open(name: CString, oflag: CInt, tflag: CInt): CInt =
+    extern
+
+  def shm_open(name: CString, oflag: CInt, mode: mode_t): CInt = extern
+
+  def shm_unlink(name: CString): CInt = extern
+
+  // Types
+
+  type posix_typed_mem_info = CStruct1[CSize]
```

Comment:
I get it that field values should be the biggest possible for each platform but I don't understand how is handled the unfixed struct size.